crates/db_schema/src/impls/post.rs

```diff
@@ -302,34 +302,38 @@ impl Saveable for PostSaved {
   }
 }
 
-#[async_trait]
-impl Readable for PostRead {
-  type Form = PostReadForm;
-  async fn mark_as_read(
+impl PostRead {
+  pub async fn mark_as_read(
     pool: &mut DbPool<'_>,
-    post_read_form: &PostReadForm,
+    post_ids: Vec<PostId>,
+    person_id: PersonId,
   ) -> Result<Self, Error> {
-    use crate::schema::post_read::dsl::{person_id, post_id, post_read};
+    use crate::schema::post_read::dsl::post_read;
     let conn = &mut get_conn(pool).await?;
+
+    let forms = post_ids
+      .into_iter()
+      .map(|post_id| PostReadForm { post_id, person_id })
+      .collect::<Vec<PostReadForm>>();
     insert_into(post_read)
-      .values(post_read_form)
-      .on_conflict((post_id, person_id))
-      .do_update()
-      .set(post_read_form)
+      .values(forms)
+      .on_conflict_do_nothing()
```

Comment:
I'm wondering how this works with multiple values. Say if one post was *already read*, so it can't insert, but the others don't have rows yet. 

Would be a good thing to add to the unit test if you haven't already.

Reply:
Would be too complex to make a test for such a minor case. Besides the postgres docs say that on conflict is done "for each individual row proposed for insertion" so there should be no problem.

https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT